From: Richard Acayan <mailingradian@gmail.com>
Date: Thu, 29 Sep 2022 19:50:28 -0400
Subject: [PATCH] arm64: dts: qcom: pm660: add fuel gauge

The PM660 has the same fuel gauge as PMI8998. Add support for
PM660 battery monitoring.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 arch/arm64/boot/dts/qcom/pm660.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/pm660.dtsi b/arch/arm64/boot/dts/qcom/pm660.dtsi
index 156b2dd..8d96776 100644
--- a/arch/arm64/boot/dts/qcom/pm660.dtsi
+++ b/arch/arm64/boot/dts/qcom/pm660.dtsi
@@ -197,6 +197,14 @@ channel@85 {
 			};
 		};
 
+		pm660_fg: fuel-gauge@4000 {
+		        compatible = "qcom,pmi8998-fg";
+		        reg = <0x4000>;
+		        interrupts = <0x0 0x40 0x3 IRQ_TYPE_EDGE_RISING>;
+		        interrupt-names = "soc-delta";
+		        status = "disabled";
+		};
+
 		pm660_rradc: adc@4500 {
 			compatible = "qcom,pm660-rradc";
 			reg = <0x4500>;
